src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509CertificateLoader.Unix.cs

```diff
@@ -619,4 +618,73 @@ internal void Dispose()
             }
         }
     }
+
+    internal abstract class Pkcs12Key : IDisposable
```

Comment:
I think we should just implement AsymmetricAlgorithm at this point, there doesn't seem enough benefit not doing it at this point. At least here you don't seem to need any new public APIs but IMO implementing will just make it easier for everyone

Reply:
I’m pretty firmly against it.

1. It would only very minimally help this pull request. We still have quite a number of type tests `if (alg is RSAOpenSsl)`, `if (alg is DSAOpenSsl)`, etc. Looking at it from a fresh perspective, all it would really do is make the SubjectPublicKeyInfo export easier.
2. Implementing AsymmetricAlgorithm significantly complicates implementation of the actual algorithm. It makes the algorithm instances mutable and it makes PKCS#8/SPKI import and export more difficult to support for .NET Framework because AsymmetricAlgorithm on .NET Framework doesn’t have any of the key loading methods.

Since it doesn’t help consumers of the algorithm, and it doesn’t help with the implementation of the algorithm, I guess I don’t see much of a reason to do it.